app/controllers/posts_controller.rb

```diff
@@ -31,6 +33,20 @@ def create
     end
   end
 
+  def destroy
+    @post = Post.find(params[:id])
+    Comment.where(post_id: @post.id).destroy_all
+    Like.where(post_id: @post.id).destroy_all
+
+    @post.destroy
+    flash[:success] = 'You deleted this post'
+
+    # Decrement the posts_counter for the user
+    @user = User.find(params[:user_id])
+    @user.update(posts_counter: @user.posts_counter - 1)
+    redirect_to user_path(@post.author), notice: 'Post Deleted!'
```

Comment:
- [OPTIONAL] You can refer to this comment for decreasing the `posts_counter`. 💻